src/rust_wgpu_backend/codegen.rs

```diff
@@ -762,70 +793,96 @@ impl<'program> CodeGen<'program> {
             &input_storage_types[0..captures.len()],
             &input_storage_types[captures.len()..],
             &output_storage_types[0..],
-        );
+        )
     }
 
+    /// Compiles the current funclet and returns a tuple for
+    /// any unhandled split points, which will be either returns or schedule calls.
+    /// # Arguments
+    /// * `current_funclet_id` - The funclet to compile
+    /// * `current_output_node_results` - The output node results of the funclet which are the
+    ///   inputs to the next funclet
+    /// * `pipeline_context` - The pipeline context
+    /// * `traversal_state_stack` - The traversal state stack, which will be pushed to if we need to
+    ///   process an inlined funclet after the current funclet
+    /// * `default_join_point_id_opt` - The default join point id to jump to after the current funclet
+    ///   is compiled
+    /// * `func_inline` - Whether the current funclet is being inlined as a call
+    /// * `branch_inline` - Whether the current funclet is being inlined as a branch
+    /// # Returns
+    /// * None for no unhandled split points, otherwise, a tuple of the
+    ///     unhandled split point node results and a boolean indicating whether
+    ///     the split point is a call
     fn process_current_funclet(
         &mut self,
         current_funclet_id: ir::FuncletId,
-        current_output_node_results: Box<[NodeResult]>,
+        current_output_node_results: &[NodeResult],
```

Comment:
I realized it might be easier to just comment directly -- this change had me a tad concerned, and I was curious why it was necessary?

Reply:
Basically, the `process_current_funclet` doesn't need to mutate the arguments anymore and follows a pattern where it only reads its arguments and then pushes the updated state onto the `traversal_state_stack`. This pattern isn't followed for the `default_join_point_id` since it was already like this and when I tried to explicitly split up the difference between processing the current funclet and mutating to prepare for the next funclet, things broke.